packages/inventory/src/Stocktake/api/hooks.ts

```diff
@@ -53,8 +53,15 @@ const useStocktakeNumber = () => {
 export const useStocktake = (): UseQueryResult<StocktakeFragment> => {
   const stocktakeNumber = useStocktakeNumber();
   const api = useStocktakeApi();
-  return useQuery(api.keys.detail(stocktakeNumber), () =>
-    api.get.byNumber(stocktakeNumber)
+  return useQuery(
+    api.keys.detail(stocktakeNumber),
+    () => api.get.byNumber(stocktakeNumber),
+    // Don't refetch when the edit modal opens, for example. But, don't cache data when this query
+    // is inactive. For example, when navigating away from the page and back again, refetch.
+    {
+      refetchOnMount: false,
+      cacheTime: 0,
```

Comment:
without this, whenever the hook `useStocktake()` is mounted, it will refetch and It's used a decent amount! An alternative is to make another hook, but I think this works well, though having the stocktake refetch on the modal opening can sometimes be good. Can revert this or change it up if it's concerning